usr.sbin/periodic/etc/daily/801.trim-zfs

```diff
@@ -0,0 +1,60 @@
+#!/bin/sh
+#
+#
+
+# If there is a global system configuration file, suck it in.
+#
+
+if [ -r /etc/defaults/periodic.conf ]
+then
+    . /etc/defaults/periodic.conf
+    source_periodic_confs
+fi
+
+case "$daily_trim_zfs_enable" in
+    [Yy][Ee][Ss])
+	echo
+	echo 'Trimming of zfs pools:'
+
+	if [ -z "${daily_trim_zfs_pools}" ]; then
+		daily_trim_zfs_pools="$(zpool list -H -o name)"
+	fi
+
+	rc=0
+	for pool in ${daily_trim_zfs_pools}; do
+		# sanity check
+		_status=$(zpool list -Hohealth "${pool}" 2> /dev/null)
+		if [ $? -ne 0 ]; then
+			rc=2
+			echo "   WARNING: pool '${pool}' specified in"
+			echo "            '/etc/periodic.conf:daily_trim_zfs_pools'"
+			echo "            does not exist"
+			continue
+		fi
+		case ${_status} in
+		FAULTED)
+			rc=3
+			echo "Skipping faulted pool: ${pool}"
+			continue ;;
+		UNAVAIL)
+			rc=4
+			echo "Skipping unavailable pool: ${pool}"
+			continue ;;
+		esac
+
+		if ! zpool status "${pool}" | grep -q '(trimming)'; then
+			echo "    starting trim of pool '${pool}'"
+			zpool trim "${pool}"
+			[ $rc -eq 0 ] && rc=1
```

Comment:
I don't understand this line

Reply:
this is to indicate informational output from the script (exit 1) but only if we didn't already encounter a more serious error (exit 3 or 4).  i copied this from 800.scrub-zfs, but it seems to be the method that most periodic scripts use to do this (or at least roughly similar).